Sources/EdgeNetworkHandlers/EdgeHit.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2020 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import AEPCore
+import Foundation
+
+/// Struct which represents an Identity hit
+struct EdgeHit: Codable {
+    /// URL to be requested for this Edge hit
+    let url: URL
+
+    /// The `EdgeRequest` for the corresponding hit
+    let request: EdgeRequest
+
+    /// Request headers for this `EdgeHit`
+    let headers: [String: String]
```

Comment:
I'm concerned with computing the headers later, specifically because that shared state of assurance or something else could change from when this hit is queued vs when it is actually processed.

Reply:
If there's no concern with computing the headers later, we can easily move this into the EdgeHitProcessor.

Reply:
Taking into consideration that these hits can be queued for longer period of time, you will want to send the attach the latest connected session. By the time the event is processed you may be either disconnected from Assurance or using a different session, so the headers needs to be updated.

Reply:
Moved the header generation into the hit processor.